### DISCUSSION TITLE
Expression evaluation

### DISCUSSION THREAD

**Discussion by wieczorek1990** [oracle/truffleruby#3496] [General]
https://github.com/oracle/truffleruby/blob/master/lib/truffle/weakref.rb#L50

Should we use something like:

```ruby
expression_a = a(...)
expression_b(expression_a)
```

Rather than what is there now:
```
expression_b(a(...))
```

**Comment by eregon** [comment]
Why?

**Comment by wieczorek1990** [comment]
I basically think of it as naming the expression results for readablity.

W dniu pon., 18.03.2024 o 16:12 Benoit Daloze ***@***.***>
napisał(a):

**Reply by eregon** [reply]
I don't think it would help readability here.
It's certainly not something to apply in general, it depends a lot on the context whether an extra local variable helps readability.

**Comment by wieczorek1990** [comment]
I think it helps complexity destructuring too.

W dniu pon., 18.03.2024 o 17:45 Benoit Daloze ***@***.***>
napisał(a):

**Comment by wieczorek1990** [comment]
I am amending the example for correctness:

```ruby
expression_a_result = expression_a(a_expression_parameters...)
expression_b_result = expression_b(expression_a_result)
expression_b_result
```

versus:

```ruby
expression_b(expression_a(a_expression_parameters...))
```

I know the argument against are interpreters, compilers without chaining of expressions.
